infra/modules/azure_api_management/variables.tf

```diff
@@ -14,12 +14,12 @@ variable "environment" {
     location        = string
     domain          = optional(string)
     app_name        = string
-    instance_number = string
+    instance_number = number
   })
 
   validation {
-    condition     = length("${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : replace(var.environment.domain, "-", "")}${var.environment.app_name}-apim-${var.environment.instance_number}") <= 50
-    error_message = "Azure API Management name must contain between 1 and 50 characters. Current value is \"${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : var.environment.domain}${var.environment.app_name}-apim-${var.environment.instance_number}\""
+    condition     = length("${var.environment.prefix}${var.environment.env_short}reg${var.environment.domain == null ? "" : replace(var.environment.domain, "-", "")}${var.environment.app_name}-apim-00") <= 50
```

Comment:
where's instance number gone?

Reply:
it's just the validate for the number of characters, since now instance number is a `number` it may not be 2 digits, but the string value that the suffix assumes is always 2 digits, so to have the same result just put 2 digits, I used `00` symbolically, the possible values are all those from 01 to 99